### PR TITLE
chore: replace game_scanner usage with winreg

### LINKED ISSUE #101: Replace `game-scanner` library with simple call to Windows registry
With #99 we use [`game-scanner`](https://crates.io/crates/game-scanner) as a library to get Origin Titanfall2 install on Windows. However this library adds quite a bunch of dependencies increasing build time.

@mrsteyk suggested using [`winreg`](https://crates.io/crates/winreg) to query Windows registry directly instead.

### PR DESCRIPTION
**Pull request by Jan200101**
closes #101

Not compiled on Windows (I don't have a setup for that, so someone else needs to do it).

Error message was updated to reflect this works on Origin and the EA App (should the text be changed to only mention the EA App, since Origin is considered defunct?), though the function name was not updated.

Moved winapi and winreg dependencies into the windows only dependencies, since other systems don't make use of them.

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Currently also don't have a working Windows setup rn to test. Will do in a day or two hopefully.

File: src-tauri/src/platform_specific/windows.rs [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-                        Err(err) => {
-                            log::warn!("{}", err);
-                            continue; // Not a valid game path
-                        }
+    #[cfg(target_os = "windows")]
+    {
+        let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+        match hklm.open_subkey("SOFTWARE\\Respawn\\Titanfall2") {
+            Ok(tf) => {
+                let game_path_str: String = tf.get_value("Install Dir")?;
```
Comment (Jan200101) [inline src-tauri/src/platform_specific/windows.rs:16]: the type that `get_value` returns depend on the type that is expected, so we cannot easily set the type of game_path_str to something like `Result<String>` without the function returning a `Result<Result<String>>`.

The curent way its done is a bit ugly but it does the job (plus what are the chances someone has the registry subkey but not the value :clueless:)

**Review by Alystrasz — APPROVED** [review]
Confirmed working on Windows 10.
FlightCore found the path of my Steam install through registry successfully.

**Comment by GeckoEidechse** [comment]
> FlightCore found the path of my Steam install through registry successfully.

Given that this affects only EA App. We should still test on a setup that only has EA App and no Steam. In particular, detecting Steam install is done via `library.vdf` and not registry.

**Comment by Jan200101** [comment]
I'll setup a clean VM and test a fresh install from both Origin and the EA App.

**Comment by Alystrasz** [comment]
My bad!

**Comment by Jan200101** [comment]
Fresh Install of Titanfall 2 from the EA App

image

I'll test Origin later today with [Fuck_off_EA_App](https://github.com/p0358/Fuck_off_EA_App)

**Comment by GeckoEidechse** [comment]
> Fresh Install of Titanfall 2 from the EA App

You're awesome, thanks <3



> I'll test Origin later today with [Fuck_off_EA_App](https://github.com/p0358/Fuck_off_EA_App)

Wouldn't mind but honestly not necessary. They should behave the same and tbh I'm not gonna consider it any more. Compared to EA App Origin is a minority and those few that still use it can probably also figure out how to manually set a path should it somehow fail ^^

File: src-tauri/src/platform_specific/windows.rs (hunk 1 above)
Comment (GeckoEidechse) [reply src-tauri/src/platform_specific/windows.rs:16]: I can probably figure out a way to improve this but given how many dependencies we can slash by removing game scanner I'm fine with merging as is ^^

**Comment by Jan200101** [comment]
On Origin (with an incomplete install. Why doesn't the EA App have this feature? its really cool!)

image

**Pull request merged**